### PR TITLE
Feature/qr

### PR DESCRIPTION
**Pull request by Tougee**

### CODE REVIEW

File: app/src/main/java/one/mixin/android/ui/url/UrlInterpreterActivity.kt [hunk 1]
```diff
@@ -86,4 +91,25 @@ fun isMixinUrl(url: String): Boolean {
     } else {
         return false
     }
+}
+
+fun openUrl(url: String, conversationId: String?, supportFragmentManager: FragmentManager) {
+    if (url.startsWith(Constants.MIXIN_TRANSFER_PREFIX, true)) {
+        val userId = url.substring(url.lastIndexOf('/') + 1)
```
Comment (crossle) [inline app/src/main/java/one/mixin/android/ui/url/UrlInterpreterActivity.kt]: Should use Url.parse url

File: app/src/main/res/drawable/ic_contacts_receive.xml [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+</vector>
```
Comment (crossle) [inline app/src/main/res/drawable/ic_contacts_receive.xml]: Use [avocado](https://github.com/alexjlockwood/avocado) optimization the vector

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -59,6 +59,10 @@
                 <data
                     android:host="users"
                     android:scheme="mixin"/>
+
+                <data
+                    android:host="transfer"
+                    android:scheme="mixin"/>
```
Comment (crossle) [inline app/src/main/AndroidManifest.xml:65]: Forgot the Activity parse this uri?

File: app/src/main/java/one/mixin/android/ui/contacts/ContactsAdapter.kt [hunk 4]
```diff
@@ -34,7 +34,7 @@ class ContactsAdapter(val context: Context, var users: List<User>, var friendSiz
     private var mHeaderView: View? = null
     private var mFooterView: View? = null
     private var mContactListener: ContactListener? = null
-    private var mSelf: User? = null
+    var self: User? = null
```
Comment (crossle) [inline app/src/main/java/one/mixin/android/ui/contacts/ContactsAdapter.kt]: self? it's like a keyword...

**Comment by Tougee** [comment]
All fixed

**Pull request merged**